### PR TITLE
Issue 23: Display public IP address

### PR DESCRIPTION
**Pull request by shivangswain**
Added a method to fetch and display public IP address in the server verbose

Linked Issue: #23

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 from prompt_toolkit import print_formatted_text, HTML
 from websockets.exceptions import ConnectionClosedError
 from utils.helper_display import HelperDisplay
+from requests import get
```
Comment (t0xic0der) [inline main.py:5]: `requests` is quite a heavy module to import for task as small as this. Also, we cannot just assume that the users have `curl` or `wget` binaries installed on their distribution which makes it a difficult point to discuss about.

File: main.py [hunk 2]
```diff
@@ -28,6 +29,10 @@ def getallus(chatroom):
     return userlist
 
 
+def ipaddress():
+    return get('https://api64.ipify.org').text
```
Comment (t0xic0der) [inline main.py:33]: This does a great job in retrieving an IPv6 address as evidenced in the screenshot I have attached just below.

image

But when it comes to IPv4, it still ends up fetching the IPv6 address.

image

File: main.py [hunk 3]
```diff
@@ -107,9 +114,10 @@ def mainfunc(chatport, netprotc):
         elif netprotc == "ipprotv4":
             print_formatted_text(HTML("[" + obtntime() + "] " + "<b>SNCTRYZERO</b> > <green>IP version : 4</green>"))
             netpdata = "0.0.0.0"
+        print_formatted_text(HTML("[" + obtntime() + "] " + "<b>SNCTRYZERO</b> > <green>IP address : " + ipaddress() + "</green>"))
         servenow(netpdata, chatport)
     except OSError:
-        print_formatted_text(HTML("[" + obtntime() + "] " + "<b>SNCTRYZERO</b> > <red><b>The server could not be started up</b></red>"))
```
Comment (t0xic0der) [inline main.py:112]: Nice touch with the corrected message. :wink:

**Review by t0xic0der — CHANGES_REQUESTED** [review]
Please take a look at the changes I have suggested.

**Comment by t0xic0der** [comment]
Any updates, @shivangswain?

**Comment by shivangswain** [comment]
Yeah, changing the whole implementation to use urllib3. ETA: 30 mins.

**Pull request closed without merging**